src/mongocxx/result/insert_many.hpp

```diff
@@ -59,11 +57,15 @@ class MONGOCXX_API insert_many {
     ///
     /// @return The values of the _id field for inserted documents.
     ///
-    std::map<std::size_t, bsoncxx::document::element> inserted_ids();
+    id_map inserted_ids();
 
    private:
+    friend collection;
+
+    insert_many(result::bulk_write result, id_map inserted_ids);
```

Comment:
Yes! Thanks for pointing that out :)
I made sure to fix this in my other pull request